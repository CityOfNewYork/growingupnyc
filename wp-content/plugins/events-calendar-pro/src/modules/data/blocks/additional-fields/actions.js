/**
 * Internal dependencies
 */
import { types } from '@moderntribe/events-pro/data/blocks/additional-fields';
import { SET_ADDITIONAL_FIELD_OUTPUT } from './types';

export const setInitialState = ( props ) => ( {
	type: types.SET_ADDITIONAL_FIELD_INITIAL_STATE,
	payload: props,
} );

export const setFieldChange = ( name ) => ( {
	type: types.SET_ADDITIONAL_FIELD_CHANGE,
	payload: {
		name,
	},
} );

export const setFieldBlur = ( name ) => ( {
	type: types.SET_ADDITIONAL_FIELD_BLUR,
	payload: {
		name,
	},
} );

export const setFieldBlurWithType = ( name, type ) => ( {
	type: `${ types.SET_ADDITIONAL_FIELD_BLUR }/${ type }`,
	payload: {
		name,
	},
} );

export const addField = ( name ) => ( {
	type: types.ADD_ADDITIONAL_FIELD,
	payload: {
		name,
	},
} );

export const removeField = ( name ) => ( {
	type: types.REMOVE_ADDITIONAL_FIELD,
	payload: {
		name,
	},
} );

export const setFieldLabel = ( name, label ) => ( {
	type: types.SET_ADDITIONAL_FIELD_LABEL,
	payload: {
		name,
		label,
	},
} );

export const setFieldIsPristine = ( name, isPristine ) => ( {
	type: types.SET_ADDITIONAL_FIELD_IS_PRISTINE,
	payload: {
		name,
		isPristine,
	},
} );

export const setFieldValue = ( name, value ) => ( {
	type: types.SET_ADDITIONAL_FIELD_VALUE,
	payload: {
		name,
		value,
	},
} );

export const appendFieldValue = ( name, value ) => ( {
	type: types.APPEND_ADDITIONAL_FIELD_VALUE,
	payload: {
		name,
		value,
	},
} );

export const removeFieldValue = ( name, value ) => ( {
	type: types.REMOVE_ADDITIONAL_FIELD_VALUE,
	payload: {
		name,
		value,
	},
} );

export const setFieldType = ( name, type ) => ( {
	type: types.SET_ADDITIONAL_FIELD_TYPE,
	payload: {
		name,
		type,
	},
} );

export const setFieldOptions = ( name, options ) => ( {
	type: types.SET_ADDITIONAL_FIELD_OPTIONS,
	payload: {
		name,
		options,
	},
} );

export const setFieldDividerList = ( name, dividerList ) => ( {
	type: types.SET_ADDITIONAL_FIELD_DIVIDER_LIST,
	payload: {
		name,
		dividerList,
	},
} );

export const setFieldDividerEnd = ( name, dividerEnd ) => ( {
	type: types.SET_ADDITIONAL_FIELD_DIVIDER_END,
	payload: {
		name,
		dividerEnd,
	},
} );

export const setFieldMetaKey = ( name, metaKey ) => ( {
	type: types.SET_ADDITIONAL_FIELD_META_KEY,
	payload: {
		name,
		metaKey,
	},
} );

export const setFieldOutput = ( name, output ) => ( {
	type: SET_ADDITIONAL_FIELD_OUTPUT,
	payload: {
		name,
		output,
	},
} );
