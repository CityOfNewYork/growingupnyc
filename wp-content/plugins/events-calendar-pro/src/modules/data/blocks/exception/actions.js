/**
 * External dependencies
 */
import { curry } from 'lodash/fp';

/**
 * Internal dependencies
 */
import * as types from './types';

export const addField = () => ( {
	type: types.ADD_EXCEPTION_FIELD,
} );

export const addException = ( payload ) => ( {
	type: types.ADD_EXCEPTION,
	payload,
} );

export const removeField = () => ( {
	type: types.REMOVE_EXCEPTION_FIELD,
} );

export const removeException = ( index ) => ( {
	type: types.REMOVE_EXCEPTION,
	index,
} );

export const editException = curry( ( index, payload ) => ( {
	type: types.EDIT_EXCEPTION,
	index,
	payload,
} ) );

export const syncException = curry( ( index, payload ) => ( {
	type: types.EDIT_EXCEPTION,
	index,
	payload,
	sync: true,
} ) );

export const syncExceptionsFromDB = payload => ( {
	type: types.SYNC_EXCEPTIONS_FROM_DB,
	payload,
} );
