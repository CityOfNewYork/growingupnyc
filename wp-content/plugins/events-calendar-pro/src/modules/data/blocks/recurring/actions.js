/**
 * External dependencies
 */
import { curry } from 'lodash/fp';

/**
 * Internal dependencies
 */
import * as types from './types';

export const addField = () => ( {
	type: types.ADD_RULE_FIELD,
} );

export const addRule = ( payload ) => ( {
	type: types.ADD_RULE,
	payload,
} );

export const removeField = () => ( {
	type: types.REMOVE_RULE_FIELD,
} );

export const removeRule = ( index ) => ( {
	type: types.REMOVE_RULE,
	index,
} );

export const editRule = curry( ( index, payload ) => ( {
	type: types.EDIT_RULE,
	index,
	payload,
} ) );

export const syncRule = curry( ( index, payload ) => ( {
	...editRule( index, payload ),
	sync: true,
} ) );

export const syncRulesFromDB = ( payload ) => ( {
	type: types.SYNC_RULES_FROM_DB,
	payload,
} );
