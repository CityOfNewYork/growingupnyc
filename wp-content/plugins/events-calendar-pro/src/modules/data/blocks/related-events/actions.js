/**
 * Internal dependencies
 */
import * as types from './types';

export const setTitle = ( title ) => ( {
	type: types.SET_RELATED_EVENTS_TITLE,
	payload: {
		title,
	},
} );

export const setDisplayImages = ( displayImages ) => ( {
	type: types.SET_RELATED_EVENTS_DISPLAY_IMAGES,
	payload: {
		displayImages,
	},
} );

export const setInitialState = ( payload ) => ( {
	type: types.SET_RELATED_EVENTS_INITIAL_STATE,
	payload,
} );
