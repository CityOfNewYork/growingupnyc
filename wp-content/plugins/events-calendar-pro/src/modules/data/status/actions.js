/**
 * Internal dependencies
 */
import * as types from './types';

export const setSeriesQueueStatus = ( payload ) => ( {
	type: types.SET_SERIES_QUEUE_STATUS,
	payload,
} );
