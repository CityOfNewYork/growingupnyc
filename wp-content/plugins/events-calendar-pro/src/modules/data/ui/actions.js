/**
 * Internal dependencies
 */
import * as types from './types';

export const toggleRepeatBlocksVisibility = () => ( {
	type: types.TOGGLE_REPEAT_EVENTS_BLOCK_VISIBILITY,
} );

export const toggleRulePanelVisibility = () => ( {
	type: types.TOGGLE_RULE_PANEL_VISIBILITY,
} );

export const hideRulePanel = () => ( {
	type: types.HIDE_RULE_PANEL,
} );

export const toggleExceptionPanelVisibility = () => ( {
	type: types.TOGGLE_EXCEPTION_PANEL_VISIBILITY,
} );

export const toggleRulePanelExpand = () => ( {
	type: types.TOGGLE_RULE_PANEL_EXPAND,
} );

export const expandRulePanel = () => ( {
	type: types.EXPAND_RULE_PANEL,
} );

export const hideExceptionPanel = () => ( {
	type: types.HIDE_EXCEPTION_PANEL,
} );

export const expandExceptionPanel = () => ( {
	type: types.EXPAND_EXCEPTION_PANEL,
} );

export const toggleExceptionPanelExpand = () => ( {
	type: types.TOGGLE_EXCEPTION_PANEL_EXPAND,
} );

