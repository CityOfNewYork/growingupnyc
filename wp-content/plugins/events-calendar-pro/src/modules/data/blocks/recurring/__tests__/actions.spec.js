/**
 * Internal Dependencies
 */
import * as actions from '../actions';

describe( 'Recurrence actions', () => {
	test( 'addField', () => {
		expect( actions.addField() ).toMatchSnapshot();
	} );

	test( 'addRule', () => {
		expect( actions.addRule( { value: 1 } ) ).toMatchSnapshot();
	} );

	test( 'removeField', () => {
		expect( actions.removeField() ).toMatchSnapshot();
	} );

	test( 'removeRule', () => {
		expect( actions.removeRule( 1 ) ).toMatchSnapshot();
	} );

	test( 'editRule', () => {
		expect( actions.editRule( 1, { value: 1 } ) ).toMatchSnapshot();
	} );

	test( 'syncRule', () => {
		expect( actions.syncRule( 1, { value: 1 } ) ).toMatchSnapshot();
	} );

	test( 'syncRulesFromDB', () => {
		expect( actions.syncRulesFromDB( { value: 1 } ) ).toMatchSnapshot();
	} );
} );
