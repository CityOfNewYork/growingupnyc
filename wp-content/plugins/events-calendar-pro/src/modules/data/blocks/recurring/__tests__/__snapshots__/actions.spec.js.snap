// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recurrence actions addField 1`] = `
Object {
  "type": "@@MT/EVENTS-PRO/ADD_RULE_FIELD",
}
`;

exports[`Recurrence actions addRule 1`] = `
Object {
  "payload": Object {
    "value": 1,
  },
  "type": "@@MT/EVENTS-PRO/ADD_RULE",
}
`;

exports[`Recurrence actions editRule 1`] = `
Object {
  "index": 1,
  "payload": Object {
    "value": 1,
  },
  "type": "@@MT/EVENTS-PRO/EDIT_RULE",
}
`;

exports[`Recurrence actions removeField 1`] = `
Object {
  "type": "@@MT/EVENTS-PRO/REMOVE_RULE_FIELD",
}
`;

exports[`Recurrence actions removeRule 1`] = `
Object {
  "index": 1,
  "type": "@@MT/EVENTS-PRO/REMOVE_RULE",
}
`;

exports[`Recurrence actions syncRule 1`] = `
Object {
  "index": 1,
  "payload": Object {
    "value": 1,
  },
  "sync": true,
  "type": "@@MT/EVENTS-PRO/EDIT_RULE",
}
`;

exports[`Recurrence actions syncRulesFromDB 1`] = `
Object {
  "payload": Object {
    "value": 1,
  },
  "type": "@@MT/EVENTS-PRO/SYNC_RULES_FROM_DB",
}
`;
